### PR TITLE
[MAINTENANCE] Remove kwarg_lookup_dict

### PR DESCRIPTION
**Pull request by tyler-hoffman**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/core/expectation_configuration.py [hunk 1]
```diff
@@ -149,792 +149,6 @@ class ExpectationConfiguration(SerializableDictDot):
         ValueError: If a `domain_type` cannot be determined.
     """
 
-    kwarg_lookup_dict: ClassVar[Mapping[str, KWargDetailsDict]] = {
```
Comment (Kilo59) [inline great_expectations/core/expectation_configuration.py:152]: If we are getting rid of this we can probably delete the `KWargDetailsDict` class as well.

File: great_expectations/data_asset/data_asset.py
Comment (Kilo59) [inline great_expectations/data_asset/data_asset.py:1]: Maybe not in this PR but...

Can we delete this whole module?
This plus the old BDS Datasource classes/modules?

File: great_expectations/dataset/pandas_dataset.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def inner_wrapper(  # noqa: PLR0913, PLR0912
                 nonnull_values[boolean_mapped_success_values == False].index
             )
 
-            if "output_strftime_format" in kwargs:
+            if kwargs.get("output_strftime_format") is not None:
```
Comment (tyler-hoffman) [inline great_expectations/dataset/pandas_dataset.py:121]: We now get the keys for null values, e.g. kwargs can look like {"output_strftime_format": None}`

File: great_expectations/core/expectation_configuration.py (hunk 1 above)
Comment (tyler-hoffman) [reply great_expectations/core/expectation_configuration.py:152]: There's still one other use of it. I think that can go as well, but it effects a number of other methods in this file, so I might keep it for a followup PR.

File: great_expectations/data_asset/data_asset.py
Comment (tyler-hoffman) [reply great_expectations/data_asset/data_asset.py:1]: Seems like a good idea. I don't want to take on more scope in this PR, but I'll add a ticket for us to do it.

File: great_expectations/expectations/core/expect_column_values_to_be_in_type_list.py [hunk 3]
```diff
@@ -105,6 +105,7 @@ class ExpectColumnValuesToBeInTypeList(ColumnMapExpectation):
         [expect_column_values_to_be_of_type](https://greatexpectations.io/expectations/expect_column_values_to_be_of_type)
     """
 
+    condition_parser: Union[str, None] = "pandas"
```
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_be_in_type_list.py:108]: Not necessary here but do we have a select number of `condition_parsers`? We should probably use a `StrEnum` or `Literal[...]` as a follow-up ticket

**Review by cdkini — APPROVED** [review]

**Pull request merged**